root/usr/bin/nm-configure

```diff
@@ -40,12 +40,16 @@ if [[ "$1" == "primary-static" ]]; then
   gw=$4
   dns=$5
   vlanid=${6:-$KCL_FDI_VLAN_PRIMARY}
-  [[ "$vlanid" != "" ]] && ctype="vlan"
+  if [[ "$vlanid" != "" ]]; then
+    ctype="vlan"
+    vlan_section="[vlan]\nid=$vlanid"
+  else
+    vlan_section=""
+  fi
   script=/etc/NetworkManager/system-connections/primary
   cat > $TMP_CFG <<EONS
 [connection]
 id=primary
-uuid=$uuid
```

Comment:
New NM hates `uuid=` when uuid prog was missing, it works without UUID so removing completely.